[Request] (typescript) Do not require downstream users to add "dom" to their TS lib declaration

**Is your request related to a specific problem you're having?**

[TypeDoc](https://typedoc.org/) uses highlight.js to highlight code samples in documentation, I recently tried to upgrade from version 10.0.0 to 10.1.2. Between these versions, the typings were changed to use `HTMLElement`, which is only defined if the downstream user's tsconfig specifies "dom" in their lib declaration. Since TypeDoc runs purely in node, without access to the browser API, I shouldn't need to specify this.

```
node_modules/highlight.js/types/index.d.ts:12:31 - error TS2304: Cannot find name 'HTMLElement'.

12     highlightBlock: (element: HTMLElement) => void
                                 ~~~~~~~~~~~

node_modules/highlight.js/types/index.d.ts:118:31 - error TS2304: Cannot find name 'HTMLElement'.

118 type HighlightedHTMLElement = HTMLElement & {result?: object, second_best?: object, parentNode: HTMLElement}
                                  ~~~~~~~~~~~

node_modules/highlight.js/types/index.d.ts:118:97 - error TS2304: Cannot find name 'HTMLElement'.

118 type HighlightedHTMLElement = HTMLElement & {result?: object, second_best?: object, parentNode: HTMLElement}
 ```

**The solution you'd prefer / feature you'd like to see added...**

Fixing this should be relatively easy. TypeScript doesn't have [forward declarations for types](https://github.com/microsoft/TypeScript/issues/31894) yet, but the `declare global` with an empty interface should work in this case.


**Any alternative solutions you considered...**
For now, I've pinned the version to 10.0.0, but would like to be able to upgrade.

--- a/types/index.d.ts
+++ b/types/index.d.ts
@@ -1,3 +1,6 @@
+// For TS consumers who use Node and don't have dom in their tsconfig lib, import the necessary types here.
+/// <reference lib="dom" />
+
 /* Public API */
 
 // eslint-disable-next-line

